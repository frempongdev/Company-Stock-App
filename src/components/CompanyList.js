import { useSelector } from 'react-redux';
import PropTypes from 'prop-types';

import { Link } from 'react-router-dom';
import '../styles/CompanyList.css';

const CompanyList = ({ searchTerm }) => {
  const { companies } = useSelector((state) => state.companies);

  const filteredCompanies = companies.filter((company) => (
    company.companyName.toLowerCase().includes(searchTerm.toLowerCase())
       || company.symbol.toLowerCase().includes(searchTerm.toLowerCase())
  ));
  return (
    <div className="company-wrapper">

      {[...filteredCompanies]
        .sort((a, b) => b.price - a.price)
        .map((company, index) => (company.isLoading ? (
          <div className="company-box" key={company.symbol}>
            <span className="load-message">loading Company details ... </span>
          </div>
        )
          : (
            <>
              <div className="company-box" key={company.symbol}>
                <div className="box-top">
                  <h3 className="rank">{index + 1}</h3>
                  <Link to={`/Profile/${company.symbol}`}>
                    <button type="button" className="back">
                      more
                      {' > '}
                    </button>
                  </Link>
                </div>
                <h3 className="company-name">{company.symbol}</h3>
                <span className="stock-price">
                  $
                  {company.price}
                </span>
                <h5 className="company-name">{company.companyName}</h5>
              </div>
            </>
          )
        ))}
    </div>
  );
};

CompanyList.propTypes = {
  searchTerm: PropTypes.string,
};

CompanyList.defaultProps = {
  searchTerm: '',
};
export default CompanyList;
