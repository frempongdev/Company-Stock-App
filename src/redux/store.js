import { configureStore } from '@reduxjs/toolkit';
import companiesReducer from './Companies/CompaniesSlice';

const store = configureStore({
  reducer: {
    companies: companiesReducer,
  },
});

export default store;
