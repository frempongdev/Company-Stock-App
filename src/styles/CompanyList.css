.company-wrapper {
    display: grid;
    grid-template-columns: 1fr 1fr;
}

.company-box {
    border: 1px solid grey;
}

.box-top {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0%;
}

.rank {
    margin: 0%;
}