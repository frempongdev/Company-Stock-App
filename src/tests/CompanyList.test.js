import { Provider, useSelector } from 'react-redux';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import CompanyList from '../components/CompanyList';
import store from '../redux/store';

describe('CompanyList Snapshot', () => {
  it('renders CompanyList', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <CompanyList />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

jest.mock('react-redux');

describe('CompanyList', () => {
  beforeEach(() => {
    useSelector.mockImplementation((callback) => callback({
      companies: {
        companies: [
          {
            symbol: 'AAPL',
            companyName: 'Apple Inc.',
            price: 145.44,
            isLoading: false,
          },
          {
            symbol: 'AMZN',
            companyName: 'Amazon.com Inc.',
            price: 3362.02,
            isLoading: false,
          },
        ],
      },
    }));
  });

  afterEach(() => {
    useSelector.mockClear();
  });

  it('renders an h3 element with the classname "company-name"', () => {
    const { getByRole } = render(
      <BrowserRouter>
        <CompanyList searchTerm="am" />
      </BrowserRouter>,
    );
    const companyElement = getByRole('heading', { name: /amzn/i });
    expect(companyElement).toHaveClass('company-name');
    expect(companyElement.tagName).toBe('H3');
  });

  it('renders a span element with the classname "stock-price"', () => {
    const { getByRole } = render(
      <BrowserRouter>
        <CompanyList searchTerm="am" />
      </BrowserRouter>,
    );
    const companyElement = getByRole('heading', { name: /amzn/i });
    expect(companyElement.nextElementSibling).toHaveClass('stock-price');
    expect(companyElement.nextElementSibling.tagName).toBe('SPAN');
  });
});
