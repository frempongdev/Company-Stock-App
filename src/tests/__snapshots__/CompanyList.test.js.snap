// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyList Snapshot renders CompanyList 1`] = `null`;
